### ISSUE TITLE
The learning rate of the first epoch is 10x of args.lr

### ISSUE THREAD

**Issue by automan000** [KaiyangZhou/deep-person-reid#20]
Hi. Great code.
I have one question about the scheduler. The initial scheduler.last_epoch =-1. And in Python3, -1 // 60 = -1, which means the actual learning rate of the first epoch is 10 times of args.lr. I think the correct code sequence is
```python
for epoch in range(args.max_epoch):
scheduler.step()
train(...)
validate(...)
```

**Comment by KaiyangZhou** [comment]
Hi,
`step()` first does `self.last_epoch + 1` so the starting epoch is 0 instead of -1. Pls refer to [here](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L45).

**Comment by automan000** [comment]
You are right. I start to track the learning rate before the initialization of scheduler accidentally.
